src/WebJobs.Script/Workers/WorkerConcurrencyManager.cs

```diff
@@ -131,10 +131,9 @@ internal async void OnTimer(object sender, System.Timers.ElapsedEventArgs e)
                 {
                     if (_functionInvocationDispatcher is RpcFunctionInvocationDispatcher rpcDispatcher)
                     {
-                        // Check memory
-                        var currentChannels = (await rpcDispatcher.GetAllWorkerChannelsAsync()).Select(x => x.WorkerProcess.Process.PrivateMemorySize64);
-                        if (IsEnoughMemoryToScale(Process.GetCurrentProcess().PrivateMemorySize64,
-                            currentChannels,
+                        var allWorkerChannels = await rpcDispatcher.GetAllWorkerChannelsAsync();
+                        if (CanScale(allWorkerChannels) && IsEnoughMemoryToScale(Process.GetCurrentProcess().PrivateMemorySize64,
```

Comment:
Should we not check for feature flag here - scale only when intelligent concurrent flag is enabled?